### ISSUE TITLE
When || operator in front of $(...) in an assignment statement, SC2091 will be claimed.

### ISSUE THREAD

**Issue by Sociosarbis** [koalaman/shellcheck#1839]
- RuleID: SC2091: 
- My shellcheck version (0.70): 
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

**My case:**
```shell
remote_repos=$branch_name || $(git branch -a | grep 'remotes/origin/\S*' | sed -E 's/.*remotes\/origin\/(\S*).*/\1/' | grep -wv 'HEAD'| tr '\n' ' ')
```
When $(...) is not placed right after the equal sign in the assignment statement, SC2091 will be claimed.

Although there is a `||` operator in front of it, but I think this line of code is stlll an assignment statement and would not cause the problem showed in SC2091.(May be.I am not sure😄)

**Comment by Nightfirecat** [comment]
SC2091 is being raised properly here because your code doesn't do what you think it does. If `$branch_name` is empty or unset, `remote_repos` will still be set to its value. Everything after the `||` is not doing anything.

**Comment by Sociosarbis** [comment]
Oh, I tested what you said just now.
.And It's true that the value before `||` whether empty or not is always picked for assignment in shell.
As I am not so familiar with shell, I can't realised it at first.
Thanks for pointing this out.🙇‍
